### PR TITLE
[3.6] Upgrade to Hibernate ORM 6.2.18.Final

### PR DESCRIPTION
**Pull request by gsmet**
Let's try to get 6.2.18.Final for the Quarkus 3.6.5 I will release tomorrow.

### CODE REVIEW

File: bom/application/pom.xml [hunk 1]
```diff
@@ -97,7 +97,7 @@
         <classmate.version>1.5.1</classmate.version>
         <!-- When updating, align bytebuddy.version to Hibernate needs as well (just below),
              as well as hibernate-orm.version-for-documentation in docs/pom.xml -->
-        <hibernate-orm.version>6.2.13.Final</hibernate-orm.version>
+        <hibernate-orm.version>6.2.18.Final</hibernate-orm.version>
         <bytebuddy.version>1.14.7</bytebuddy.version> <!-- Version controlled by Hibernate ORM's needs -->
         <hibernate-commons-annotations.version>6.0.6.Final</hibernate-commons-annotations.version> <!-- version controlled by Hibernate ORM -->
         <hibernate-reactive.version>2.0.6.Final</hibernate-reactive.version>
```
Comment (yrodiere) [inline bom/application/pom.xml:103]: Maybe upgrade to HR 2.0.7 while you're at it? It was supposedly the version to use with ORM 6.2.13.Final...

**Review by yrodiere — APPROVED** [review]

File: bom/application/pom.xml (hunk 1 above)
Comment (rsvoboda) [reply bom/application/pom.xml]: Isn't https://github.com/hibernate/hibernate-reactive/commit/163eee7a4e9b6ee6b4ce183c735b43fbec795ae2 problematic as 3.6 branch is using Vert.x 4.4 - https://github.com/quarkusio/quarkus/blob/3.6/bom/application/pom.xml#L121

Comment (yrodiere) [reply bom/application/pom.xml]: @DavideD Any idea where this upgrade comes from? Seems strange to upgrade to a new vertx version in a micro of Hibernate Reactive...

Comment (DavideD) [reply bom/application/pom.xml]: Sorry, which upgrade?  Hibernate Reactive uses Vert.x 4.4 since 2.0.0

Comment (yrodiere) [reply bom/application/pom.xml]: @DavideD The one @rsvoboda linked just above: https://github.com/hibernate/hibernate-reactive/commit/163eee7a4e9b6ee6b4ce183c735b43fbec795ae2
It's on the 2.0 branch, oddly, and also included in 2.0.7.Final.

Comment (DavideD) [reply bom/application/pom.xml]: Probably a mistake on my part.

**Comment by gsmet** [comment]
I removed the Hibernate Reactive upgrade for now.

**Pull request merged**